src/models/Contact.ts

```diff
@@ -1,59 +1,62 @@
-import { getDomainByPubKey } from 'api/RIFNameService';
-import Message from 'models/Message';
-import Multiaddr from 'multiaddr';
-import { PeerInfo } from 'peer-info';
-import { createPeerIdFromPublicKey, createPeerInfo } from 'rif-communications';
-import { IUserInfo } from 'types';
+import { getDomainByPubKey } from 'api/RIFNameService'
+import Message from 'models/Message'
+import Multiaddr from 'multiaddr'
+import { PeerInfo } from 'peer-info'
+import { createPeerIdFromPublicKey, createPeerInfo } from 'rif-communications'
+import { IUserInfo } from 'types'
 
 export interface IContactParams {
-  rnsName?: string;
-  publicKey: string;
-  multiaddr?: string;
-  chat?: Message[];
+  rnsName?: string
+  publicKey: string
+  multiaddr?: string
+  chat?: Message[]
 }
 
 export default class Contact implements IUserInfo {
   public static new = async (contactParams: IContactParams) => {
-    const contact = new Contact(contactParams);
-    const { publicKey, multiaddr } = contactParams;
+    const contact = new Contact(contactParams)
+    const { publicKey, multiaddr } = contactParams
 
-    const peerId = await createPeerIdFromPublicKey(publicKey);
-    contact.peerInfo = await createPeerInfo(peerId);
+    const peerId = await createPeerIdFromPublicKey(publicKey)
+    contact.peerInfo = await createPeerInfo(peerId)
     if (multiaddr) {
-      contact.peerInfo.multiaddrs.add(new Multiaddr(multiaddr));
+      contact.peerInfo.multiaddrs.add(new Multiaddr(multiaddr))
     }
 
-    return contact;
+    return contact
   };
-  public rnsName?: string;
-  public peerInfo: PeerInfo;
-  public publicKey: string;
-  public chat: Message[];
+  public rnsName?: string
+  public peerInfo: PeerInfo
+  public publicKey: string
+  public chat: Message[]
+
+  public get lastMessage() {
+    const chat = this.chat.filter(c => !c.isSync)
```

Comment:
3rd place where you filter.

Reply:
To be honest. I kinda left it because I expected that we would not want to store sync messages at all to prevent spamming the array.

Reply:
makes sense